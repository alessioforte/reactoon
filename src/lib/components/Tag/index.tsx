import React, { FC, ReactElement } from 'react';
import PropTypes from 'prop-types';
import styled, { withTheme, ThemeProvider } from 'styled-components';
import Icon from '../Icon';
import { styles, getContrastYIQ } from '../../theme';

type Props = {
  label?: string,
  icon?: string,
  action?: (e: any) => void,
  theme: any
}

const Tag: FC<Props> = ({ label, icon, action, theme }) => {

  let tagIcon: any = null;
  if (icon && action) {
    tagIcon = (
      <IconTag onClick={action} pointer>
        <Icon
          name={icon}
          size='8px'
          color={theme.colors[getContrastYIQ(theme.colors.primary)]}
        />
      </IconTag>
    )
  }
  if (icon && !action) {
    tagIcon = (
      <IconTag>
        <Icon
          name={icon}
          size='8px'
          color={theme.colors[getContrastYIQ(theme.colors.primary)]}
        />
      </IconTag>
    )
  }

  return (
    <ThemeProvider theme={theme}>
      <Badge>
        {label}{tagIcon}
      </Badge>
    </ThemeProvider>
  )
}

Tag.propTypes = {
  label: PropTypes.string,
  icon: PropTypes.string,
  action: PropTypes.func,
  theme: PropTypes.any
}

Tag.defaultProps = {
  label: '',
  theme: styles
}

export default withTheme(Tag);

const Badge = styled.div`
  color: ${props => props.theme.colors[getContrastYIQ(props.theme.colors.primary)]};
  background: ${props => props.theme.colors.primary};
  display: inline-flex;
  padding: 3px 6px;
  margin: 1px;
  border-radius: 3px;
  justify-content: center;
  align-items: center;
  font-size: 14px;
`;
const IconTag = styled.div<{pointer?: boolean}>`
  display: flex;
  align-items: center;
  margin-left: 10px;
  cursor: ${props => props.pointer ? 'pointer' : 'default'};
`
