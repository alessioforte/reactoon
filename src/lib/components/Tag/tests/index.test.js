import React from 'react';
import { render } from '@testing-library/react';
import Tag from '../';

describe('<Tag />', () => {
  it('should render and match the snapshop', () => {
    const {
      container: { firstChild }
    } = render(<Tag label='test' />);
    expect(firstChild).toMatchSnapshot();
  });
});
