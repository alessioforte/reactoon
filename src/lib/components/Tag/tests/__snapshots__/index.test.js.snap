// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should render and match the snapshop 1`] = `
<div
  class="sc-htpNat ikGEGQ"
>
  test
</div>
`;
